src/supertux/game_object_change.cpp

```diff
@@ -0,0 +1,92 @@
+//  SuperTux
+//  Copyright (C) 2024 Vankata453
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#include "supertux/game_object_change.hpp"
+
+#include "util/log.hpp"
+#include "util/reader_iterator.hpp"
+#include "util/reader_mapping.hpp"
+#include "util/writer.hpp"
+
+GameObjectChange::GameObjectChange(const std::string& name_, const UID& uid_,
+                                   const std::string& data_, const std::string& new_data_,
+                                   Action action_) :
+  name(name_),
```

Comment:
Can you do me a favour and prefix member variables with `m_`? Then you don't need to suffix `_` for the parameters.

Reply:
The reason I did that is because those are public members, and since the point of these classes is to act more like a structure containing multiple variables, it might look awkward always referring to them as `change.m_...`. If that doesn't seem like an issue to you, I could.

Reply:
Okay, even though every single cell in my body is bothered by this, I'll let it be.